datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/dataset/DatasetType.java

```diff
@@ -184,6 +192,30 @@ public List<BrowsePath> browsePaths(@Nonnull String urn, @Nonnull final QueryCon
         return BrowsePathsMapper.map(result);
     }
 
+    @Override
+    public List<Dataset> batchUpdate(@Nonnull BatchDatasetUpdateInput[] input, @Nonnull QueryContext context) throws Exception {
+        final CorpuserUrn actor = CorpuserUrn.createFromString(context.getAuthentication().getActor().toUrnStr());
+
+        final Collection<MetadataChangeProposal> proposals = Arrays.stream(input).map(updateInput -> {
+            if (isAuthorized(updateInput.getUrn(), updateInput.getUpdate(), context)) {
```

Comment:
What happens if I'm authorized to update the first dataset, but not the second?

Then we have a partial update? Getting into a partially updated state like this is dangerous - it's confusing to the end user which updates worked and which did not.

Let's instead validate authorization for each dataset in the batch, only if ALL pass authorization we apply the updte.

Reply:
If isAuthorized fails on any of the update inputs, then the method will throw an AuthorizationException - as far as I know, this should immediately exit the batchUpdate call and return to the resolver without submitting any change proposals. I can't see any dangerous side effects within the lambda, but I'm happy to deal with this in a more explicit way if you prefer.

Reply:
Wait you're totally right. Misread it!